lib/serializer.ts

```diff
@@ -435,20 +435,16 @@ export class Serializer {
 
       for (const key in modelProps) {
         if (modelProps.hasOwnProperty(key)) {
-
-          const jpath = ["responseBody"];
           const paths = this.splitSerializeName(modelProps[key].serializedName);
-          paths.forEach((item) => {
-            jpath.push(`["${item}"]`);
-          });
           // deserialize the property if it is present in the provided responseBody instance
           let propertyInstance;
-          try {
-            /*jslint evil: true */
-            propertyInstance = eval(jpath.join(""));
-          } catch (err) {
-            continue;
-          }
+          let res = responseBody;
+          // traversing the object step by step.
+          paths.forEach((item) => {
```

Comment:
would maybe just `for(const item of paths)` instead of forEach as its recommended I think for es6 and more performant